core/src/main/scala/org/apache/spark/sql/streaming/DirectKafkaStreamSource.scala

```diff
@@ -16,63 +16,135 @@
  */
 package org.apache.spark.sql.streaming
 
-import scala.reflect.ClassTag
-
-import kafka.serializer.Decoder
-
+import org.apache.kafka.common.TopicPartition
 import org.apache.spark.Logging
 import org.apache.spark.sql._
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.encoders.RowEncoder
 import org.apache.spark.sql.sources.{BaseRelation, DataSourceRegister}
 import org.apache.spark.sql.types.StructType
 import org.apache.spark.streaming.dstream.DStream
-import org.apache.spark.streaming.kafka.KafkaUtils
-import org.apache.spark.util.Utils
+import org.json4s.NoTypeHints
+import org.json4s.jackson.Serialization
+import scala.collection.mutable.HashMap
+import scala.util.control.NonFatal
+import org.apache.spark.streaming.kafka010.{ConsumerStrategies, KafkaUtils, LocationStrategies}
 
 class DirectKafkaStreamSource extends StreamPlanProvider with DataSourceRegister {
 
-  override def shortName(): String = SnappyContext.DIRECT_KAFKA_STREAM_SOURCE
+  override def shortName(): String = SnappyContext.KAFKA_STREAM_SOURCE
 
   override def createRelation(sqlContext: SQLContext,
-      options: Map[String, String],
-      schema: StructType): BaseRelation = {
+                              options: Map[String, String],
+                              schema: StructType): BaseRelation = {
     new DirectKafkaStreamRelation(sqlContext, options, schema)
   }
 }
 
 final class DirectKafkaStreamRelation(
-    @transient override val sqlContext: SQLContext,
-    opts: Map[String, String],
-    override val schema: StructType)
-    extends StreamBaseRelation(opts)
+                                       @transient override val sqlContext: SQLContext,
+                                       opts: Map[String, String],
+                                       override val schema: StructType)
+  extends StreamBaseRelation(opts)
     with Logging with Serializable {
 
-  val topicsSet = options("topics").split(",").toSet
+  val topics = options("subscribe").split(",").toList
```

Comment:
Any specific reason to change set to list. Is this dependent on order? Perhaps a LinkedHashSet will be better if it is dependent on order.

Reply:
Reverted back to using Set again as order doesn't seem to be required here. @ymahajan please let us know in case you faced any issues with the use of Set.